app/matrix_bot/eventparser.py

```diff
@@ -38,15 +39,25 @@ def is_from_userid(self, userid: str) -> bool:
     def is_from_this_bot(self) -> bool:
         return self.is_from_userid(self.matrix_client.user_id)
 
+    def is_sender_allowed(self) -> bool:
+        return self.sender_domain() in config.users_allowed_domains
+
     def room_is_direct_message(self) -> bool:
         return room_is_direct_message(self.room)
 
-    def sender_id(self):
+    def sender_id(self) -> str:
         return self.event.sender
 
     def sender_username(self) -> str:
         return self.room.users[self.event.sender].name
 
+    def sender_domain(self) -> str:
+        """
+        Sender IDs are formatted like this: "@<mail_username>-<mail_domain>:<matrix_server>
+        e.g. @john-doe-ministere_example.gouv.fr:agent.ministere_example.tchap.gouv.fr
+        """
+        return self.event.sender.split(":")[0].split("-")[-1]
+
```

Comment:
What happens if a user has "-" in its mail_username ?

Reply:
This takes the last segment after splitted by "-" so a username including a "-" won't be an issue. The issue will arise if a domain includes a "-", which is the case for "developpement-durable.gouv.fr" for example, but not yet to be an issue according to the allowed domains.

Reply:
Yes, ok you're right. let's go with the stopgap approach.